v2/decoder_user.go

```diff
@@ -61,6 +64,9 @@ func (oa v1UserClaims) Migrate() (*UserClaims, error) {
 }
 
 func (oa v1UserClaims) migrateV1() (*UserClaims, error) {
+	if oa.Max != NoLimit {
```

Comment:
I think we need to migrate, but have some sort of soft warning that fields were dropped - this would solve the issue I put above. - something like:
migrate() (*UserClaims, dropped, error)